src/content/en/fundamentals/instant-and-offline/service-worker-lifecycle/index.md

```diff
@@ -0,0 +1,678 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: This is the page description placed in the head.
+
+{# wf_updated_on: 2016-09-29 #}
+{# wf_published_on: 2016-09-29 #}
+
+# The Service Worker Lifecycle {: .page-title }
+
+{% include "web/_shared/contributors/jakearchibald.html" %}
+
+The lifecycle of the service worker is its most complicated part. If you don't know what it's trying to do and what the benefits are, it can feel like it's fighting against you. But once you know how it works, you can deliver seemless, unobtrusive updates to users, mixing together the best of web and native patterns.
+
+This is a deep dive, but the bullets at the start of each section cover most of what you need to know.
+
+## The intent
+
+The intent of the lifecycle is to:
+
+* Make offline-first possible.
+* Allow a new service worker to get itself ready without disrupting the current one.
+* Ensure an in-scope page is controlled by the same service worker (or no service worker) throughout.
+* Ensure there's only one version of your site running at once.
+
+That last one is pretty important. Without service workers, users can load one tab to your site, then later open another. This can result in two versions of your site running at the same time. Sometimes this is ok, but if you're dealing with storage you can easily end up with two tabs having very different opinions on how their shared storage should be managed. This can result in errors, or worse, data loss.
+
+Caution: Users actively dislike data loss. It causes them great sadness.
+
+## The first service worker
+
+In brief:
+
+* The install event is the first event a service worker gets, and it only happens once.
+* A promise passed to `installEvent.waitUntil()` signals the duration & success/failure of your install
+* A service worker won't receive events like "fetch" & "push" until it successfully finishes installing and becomes "active".
+* By default, a page's fetches won't go through a service worker unless the page request itself went through a service worker. So you'll need to refresh the page to see the effects of the service worker.
+* `clients.claim()` can override this default, and take control of non-controlled pages.
+
+<style>
```

Comment:
I want the iframe to maintain an aspect ratio. Also I want my browser screenshots to have a slight shadow.